compiler/plugins/target/ROCM/builtins/ukernel/argmax_ukernel.c

```diff
@@ -5,14 +5,50 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <float.h>
-#include <hip/hip_fp16.h>
-#include <hip/hip_runtime.h>
+#include <stdint.h>
+#include <string.h>
 
-extern "C" __device__ __attribute__((const)) half __ockl_wfred_max_f16(half);
-extern "C" __device__
-    __attribute__((const)) int64_t __ockl_wfred_min_i64(int64_t);
-extern "C" __device__
-    __attribute__((const)) int32_t __ockl_wfred_min_i32(int32_t);
+/*
+Bits copied from HIP headers.
+*/
+
+__attribute__((const)) _Float16 __ockl_wfred_max_f16(_Float16);
+__attribute__((const)) int64_t __ockl_wfred_min_i64(int64_t);
+__attribute__((const)) int32_t __ockl_wfred_min_i32(int32_t);
+__attribute__((const)) float __ocml_fmax_f32(float, float);
```

Comment:
`/opt/rocm/llvm/bin/llvm-dis /opt/rocm/amdgcn/bitcode/ocml.bc -o - | less` let me see that these two functions are direct wrappers around `llvm.maximum.T`, we might want to use that. (and I've heard the compiler folks make similar comments)

Reply:
So from a C perspective, I guess I just need a naive `x < y ? y : x` to get that lowering... I'll try that.

Reply:
I know that in Triton, they prefer to use ocml for some things like exp2 because it gives them the quick exp2. Would it be better to keep it as ocml_fmax in the scenario that the OCML team implements some fast max as well?